nodes/signal_shutdown.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# Simple script for shutting down the raspberry Pi at the press of a button.
+# by Inderpreet Singh
+# Adapted by Jake Rye and Bob Davis
+
+import RPi.GPIO as GPIO
```

Comment:
It looks like moving to python-periphery requires the appropriate access to the GPIO files. As a workaround access can be granted for the user running the code. 

[](https://superuser.com/questions/826124/use-sys-class-gpio-in-python-without-root-permissions)

In short, we'll need to add these two lines to the install script (something similar):

sudo echo \
   SUBSYSTEM=="gpio*", PROGRAM="/bin/sh -c \ 
       'chown -R root:gpio /sys/class/gpio && chmod -R 770 /sys/class/gpio; \
        chown -R root:gpio /sys/devices/virtual/gpio && chmod -R 770 /sys/devices/virtual/gpio'" \
>> /etc/udev/rules.d/99-com.rules